EEJT-0002/src/main/java/com/epam/esm/giftcertficate/controller/GiftCertificateController.java

```diff
@@ -0,0 +1,72 @@
+package com.epam.esm.giftcertficate.controller;
+
+import com.epam.esm.giftcertficate.service.GiftCertificateService;
+import com.epam.esm.giftcertficate.model.GiftCertificate;
+import com.epam.esm.utils.validation.DataValidation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Map;
+import java.util.Optional;
+
+@RestController
+@RequestMapping(value = "/certificates", produces = MediaType.APPLICATION_JSON_VALUE)
+public class GiftCertificateController {
+    private final GiftCertificateService giftCertificateService;
+
+    @Autowired
+    public GiftCertificateController(GiftCertificateService giftCertificateService) {
+        this.giftCertificateService = giftCertificateService;
+    }
+
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> createCertificate(@RequestBody GiftCertificate giftCertificate) {
+
+        if (DataValidation.isValidCertificate(giftCertificate)) {
+            giftCertificateService.createCertificate(giftCertificate);
+
+            return new ResponseEntity<>(Map.of("status", HttpStatus.CREATED), HttpStatus.CREATED);
+        }
+        throw new IllegalArgumentException("Something went wrong during the request, check your fields");
```

Comment:
in such case, it's better to throw specifix exception or rreturn ResponseEntity here or add validation annotations to the GiftCertificate

Reply:
I use specific ServerException ; fix it